### PR TITLE
Convert autogen.yaml to deploymentmanager-autogen spec

### PR DESCRIPTION
**Pull request by gibbleyg**
Change-Id: I9c9eea6e2f64cabc67287ea57ee1243ceebfc703

### CODE REVIEW

File: mpdev/autogen/scripts/install_protogen.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+PROTOC_ZIP=protoc-3.7.1-linux-x86_64.zip
+curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.7.1/$PROTOC_ZIP
```
Comment (huyhg) [inline mpdev/autogen/scripts/install_protogen.sh:19]: The `v3.7.1` directory should also be parameterized. Perhaps we should just parameterize the version?

```bash
PROTOC_VERSION="3.7.1"
curl -OL "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip"
```

File: mpdev/autogen/scripts/install_protogen.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+unzip -o $PROTOC_ZIP -d /usr/local bin/protoc
+unzip -o $PROTOC_ZIP -d /usr/local 'include/*'
+rm -f $PROTOC_ZIP
+
+go install google.golang.org/protobuf/cmd/protoc-gen-go
```
Comment (huyhg) [inline mpdev/autogen/scripts/install_protogen.sh:24]: nit: add a new line EOF

File: mpdev/autogen/scripts/run_protogen.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# limitations under the License.
+
+set -eu
+
+mkdir -p generated
+PROTO_DIR=java/com/google/cloud/deploymentmanager/autogen
+TMPDIR=$(mktemp -d)
+protoc -Ideploymentmanager-autogen --go_out=$TMPDIR deploymentmanager-autogen/${PROTO_DIR}/*.proto
+mv $TMPDIR/${PROTO_DIR}/* generated
+rm -rf $TMPDIR
```
Comment (huyhg) [inline mpdev/autogen/scripts/run_protogen.sh:23]: nit: add a new line EOF

File: mpdev/autogen/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+COPY scripts .
+COPY deploymentmanager-autogen deploymentmanager-autogen
+RUN ./install_protogen.sh && ./run_protogen.sh
+
+COPY . .
+RUN go build -v -o /usr/local/bin/autogen
+
+FROM alpine:latest
+COPY --from=0 /usr/local/bin/autogen /usr/local/bin/autogen
+CMD ["autogen"]
```
Comment (huyhg) [inline mpdev/autogen/Dockerfile:32]: nit: add a new line EOF

File: go.mod [hunk 5]
```diff
@@ -5,12 +5,11 @@ go 1.13
...
 	github.com/GoogleContainerTools/kpt v0.25.0
 	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e // indirect
-	github.com/golang/protobuf v1.3.5 // indirect
-	github.com/google/go-cmp v0.4.0 // indirect
 	github.com/spf13/cobra v0.0.5
 	golang.org/x/net v0.0.0-20200324143707-d3edc9973b7e // indirect
 	golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d // indirect
 	golang.org/x/sys v0.0.0-20200331124033-c3d80250170d // indirect
 	google.golang.org/appengine v1.6.5 // indirect
+	google.golang.org/protobuf v1.24.0 // indirect
```
Comment (huyhg) [inline go.mod:13]: what caused these changes? IIUIC, it should be scoped into the new `autogen` directory only

File: mpdev/autogen/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2020 Google LLC
```
Comment (huyhg) [inline mpdev/autogen/Dockerfile:1]: We probably should think through the directory structure for these type-specific containers (and refactor them afterward).

File: mpdev/autogen/Dockerfile (hunk 4 above)
Comment (gibbleyg) [reply mpdev/autogen/Dockerfile]: Done

File: mpdev/autogen/scripts/run_protogen.sh (hunk 3 above)
Comment (gibbleyg) [reply mpdev/autogen/scripts/run_protogen.sh]: Done

File: mpdev/autogen/scripts/install_protogen.sh (hunk 2 above)
Comment (gibbleyg) [reply mpdev/autogen/scripts/install_protogen.sh]: Done

File: go.mod (hunk 5 above)
Comment (gibbleyg) [reply go.mod]: Hmm this was probably added before I put the autogen in a separate `go.mod`. Will remove

**Pull request merged**